packages/block-editor/src/components/rich-text/should-dismiss-pasted-files.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Given a collection of DataTransfer files and HTML and plain text strings,
+ * determine whether the files are to be dismissed in favor of the HTML.
+ *
+ * Certain office-type programs, like Microsoft Word or Apple Numbers,
+ * will, upon copy, generate a screenshot of the content being copied and
+ * attach it to the clipboard alongside the actual rich text that the user
+ * sought to copy. In those cases, we should let Gutenberg handle the rich text
+ * content and not the screenshot, since this allows Gutenberg to insert
+ * meaningful blocks, like paragraphs, lists or even tables.
+ *
+ * @param {File[]} files File objects obtained from a paste event
+ * @param {string} html  HTML content obtained from a paste event
+ * @return {boolean}     True if the files should be dismissed
+ */
+export function shouldDismissPastedFiles( files, html /*, plainText */ ) {
+	// The question is only relevant when there is actual HTML content and when
+	// there is exactly one image file.
+	if (
+		html &&
+		files?.length === 1 &&
```

Comment:
Should the type in the doc above be adjusted?

Reply:
Adjusted how, @ellatrix?

Reply:
I see an array is expected, but `files` can be null or undefined?

Reply:
Oh, I see what you mean! I think it makes more sense to keep the signature strict (`File[]`) than to allow null/undefined values.

That said, I'm tempted to leave the optional chaining operator, just because there are different possible calling contexts for this function, including React Native components, and I feel less confident that they will always return an array (e.g. `event.nativeEvent.files`).